apiserver/facades/client/client/client.go

```diff
@@ -695,6 +695,22 @@ func (c *Client) APIHostPorts() (result params.APIHostPortsResult, err error) {
 	return result, nil
 }
 
+// APIHostPortsForAgents returns the API host/port addresses stored in state
+// and intended for use by agents based on the configured management space.
+// If no such space is configured, this will return the same as APIHostPorts().
+func (c *Client) APIHostPortsForAgents() (result params.APIHostPortsResult, err error) {
```

Comment:
As before, we shouldn't need to expose this on the server side in the Client facade.